Yafc.Parser/Data/FactorioDataDeserializer_Entity.cs

```diff
@@ -467,12 +435,38 @@ void parseEffect(LuaTable effect) {
                         projectile.placeEntities.Add(createdEntity);
                     }
                 }
-
+                break;
+            case "reactor":
+                var reactor = GetObject<Entity, EntityReactor>(name);
+                reactor.reactorNeighborBonus = table.Get("neighbour_bonus", 1f); // Keep UK spelling for Factorio/LUA data objects
+                _ = table.Get("consumption", out usesPower);
+                reactor.basePower = ParseEnergy(usesPower);
+                reactor.baseCraftingSpeed = reactor.basePower;
+                recipeCrafters.Add(reactor, SpecialNames.ReactorRecipe);
+                break;
+            case "rocket-silo":
+                goto case "furnace";
+            case "solar-panel":
+                var solarPanel = GetObject<Entity, EntityCrafter>(name);
+                solarPanel.energy = voidEntityEnergy;
+                _ = table.Get("production", out string? powerProduction);
+                recipeCrafters.Add(solarPanel, SpecialNames.GeneratorRecipe);
+                solarPanel.baseCraftingSpeed = ParseEnergy(powerProduction) * 0.7f; // 0.7f is a solar panel ratio on nauvis
+                break;
+            case "transport-belt":
+                GetObject<Entity, EntityBelt>(name).beltItemsPerSecond = table.Get("speed", 0f) * 480f;
                 break;
             case "unit-spawner":
                 var spawner = GetObject<Entity, EntitySpawner>(name);
                 spawner.capturedEntityName = table.Get<string>("captured_spawner_entity");
                 break;
+
+                // NOTE: Please add new cases in alphabetical order, even if the new case shares code with another case. That is,
+                // case "rocket-silo":
+                //    goto case "furnace";
+                // instead of 
+                // case "furnace":
+                // case "rocket-silo": // Out of order
```

Comment:
This warning is about the switch, so I think it is better to have it at the start of the switch. What do you think?

Reply:
I tend to put new cases at the end of switch blocks, but historically, the changes have been scattered through the block.